### PR TITLE
Remove SameSubnetidsValidator for awsbatch queue

### PR DESCRIPTION
**Pull request by chenwany**
Signed-off-by: chenwany <chenwany@amazon.com>

**Please See** [Git Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions)

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/src/pcluster/schemas/cluster_schema.py [hunk 1]
```diff
@@ -1139,15 +1139,15 @@ def right_scheduler_schema(self, data, **kwargs):
     @validates_schema
     def same_subnet_in_different_queues(self, data, **kwargs):
         """Validate subnet_ids configured in different queues are the same."""
-        for queues in ["slurm_queues", "aws_batch_queues"]:
-            if queues in data:
+        queues = "slurm_queues"
```
Comment (enrico-usai) [inline cli/src/pcluster/schemas/cluster_schema.py:1142]: minor: we can avoid defining a variable for this.

File: cli/src/pcluster/schemas/cluster_schema.py [hunk 2]
```diff
@@ -1139,15 +1139,15 @@ def right_scheduler_schema(self, data, **kwargs):
...
-                    return queue.networking and queue.networking.subnet_ids
+            def _queue_has_subnet_ids(queue):
+                return queue.networking and queue.networking.subnet_ids
 
-                subnet_ids = {tuple(set(q.networking.subnet_ids)) for q in data[queues] if _queue_has_subnet_ids(q)}
-                if len(subnet_ids) > 1:
-                    raise ValidationError("SubnetIds configured in different queues should be the same.")
+            subnet_ids = {tuple(set(q.networking.subnet_ids)) for q in data[queues] if _queue_has_subnet_ids(q)}
+            if len(subnet_ids) > 1:
+                raise ValidationError("SubnetIds configured in different queues should be the same.")
```
Comment (enrico-usai) [inline cli/src/pcluster/schemas/cluster_schema.py:1150]: I'd write something like: (suggested by Tim) `The SubnetIds used for all of the queues should be the same`

File: cli/src/pcluster/schemas/cluster_schema.py (hunk 1 above)
Comment (chenwany) [reply cli/src/pcluster/schemas/cluster_schema.py:1142]: queues is used in  `if queues in data:` and `for q in data[queues] if _queue_has_subnet_ids(q)}`, So I define the variable here to remove duplication

File: cli/src/pcluster/schemas/cluster_schema.py (hunk 2 above)
Comment (chenwany) [reply cli/src/pcluster/schemas/cluster_schema.py:1150]: Changed

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**